core/src/main/scala/kafka/server/checkpoints/CheckpointFile.scala

```diff
@@ -82,9 +82,8 @@ class CheckpointFile[T](val file: File, version: Int, formatter: CheckpointFileF
         line = reader.readLine()
         if (line == null)
           return Seq.empty
-        val version = line.toInt
-        version match {
-          case version =>
+        line.toInt match {
+          case `version` =>
```

Comment:
Without this change, it would always evaluate to true even if it the read version was not the same as the field `version`.